--- xboard-4.2.7/cmail.in.quote	2003-10-28 03:21:00.000000000 +0800
+++ xboard-4.2.7/cmail.in	2005-06-03 04:18:30.000000000 +0800
@@ -497,14 +497,14 @@
 	    if (@ARCHIVE) {
 		local ($date) = &get_date_from_games (@ARCHIVE) ;
 		$XBOARD_ARGS = join (' ', (("-lgf",
-					    "$ARCDIR/$PGN_GAME.$date.archive"),
-					   "-ncp",
+					    "'$ARCDIR/$PGN_GAME.$date.archive'"),
+					   @NCP_FLAGS,
 					   "-xics",
 					   @TD_FLAGS,
 					   @DEBUG_FLAGS,
 					   @UNREC_ARGS)) ;
 	    } else {
-		$XBOARD_ARGS = join (' ', (("-cmail", $PGN_GAME),
+		$XBOARD_ARGS = join (' ', (("-cmail", "'$PGN_GAME'"),
 					   @TD_FLAGS,
 					   @NCP_FLAGS,
 					   "-xics",
@@ -513,8 +513,8 @@
 	    }
 	} else {
 	    $PGN_GAME = "unknown.cmail" ;
-	    $XBOARD_ARGS = join (' ', (("-lgf", $PGN_GAME),
-				       "-ncp",
+	    $XBOARD_ARGS = join (' ', (("-lgf", "'$PGN_GAME'"),
+				       @NCP_FLAGS,
 				       "-xics",
 				       @TD_FLAGS,
 				       @DEBUG_FLAGS,
@@ -527,15 +527,15 @@
 	$PID_FILE = "$PGN_GAME.pid" ;
 	if (   (! $REUSE)
 	    || (! (   (-f $PID_FILE)
-		   && ($XBOARD_PID = `cat $PID_FILE`)
+		   && ($XBOARD_PID = `cat '$PID_FILE'`)
 		   && ("$XBOARD_PID" =~ /^\d+$/)
 		   && (kill "SIGUSR1", $XBOARD_PID)))) {
 	    print "Loading xboard for game \"$PGN_GAME\"..." ;
 #  	    system ("gdb xboard") ;
 	    system (  "{ ({ xboard $XBOARD_ARGS & } ;"
-		    . "   echo \$! > $PID_FILE ;"
+		    . "   echo \$! > '$PID_FILE' ;"
 		    . "   wait ;"
-		    . "   rm $PID_FILE) & } >$LOG_FILE 2>&1") ;
+		    . "   rm $PID_FILE) & } >'$LOG_FILE' 2>&1") ;
 	    print (  "done.\n"
 		   . "If nothing happens look for an error message in\n"
 		   . "$CMAILDIR/$LOG_FILE\n") ;
