--- xboard-4.2.7/xboard.c.xvt	2005-06-03 05:09:13.000000000 +0800
+++ xboard-4.2.7/xboard.c	2005-06-03 05:18:12.000000000 +0800
@@ -6640,7 +6640,7 @@
      Cardinal *nprms;
 {
     char buf[MSG_SIZ];
-    sprintf(buf, "xterm -e info --directory %s --directory . -f %s &",
+    sprintf(buf, "xvt -e 'info --directory %s --directory . -f %s' &",
 	    INFODIR, INFOFILE);
     system(buf);
 }
@@ -6657,7 +6657,7 @@
       name = prms[0];
     else
       name = "xboard";
-    sprintf(buf, "xterm -e man %s &", name);
+    sprintf(buf, "xvt -e 'man %s' &", name);
     system(buf);
 }
 
