--- xboard-4.2.7/backend.c.str	2009-03-28 09:22:44.000000000 +0100
+++ xboard-4.2.7/backend.c	2009-03-28 09:23:14.000000000 +0100
@@ -6581,7 +6581,7 @@
 	    fprintf(f, " ");
 	    linelen++;
 	}
-	fprintf(f, numtext);
+	fprintf(f, "%s", numtext);
 	linelen += numlen;
 
 	/* Get move */
@@ -6599,7 +6599,7 @@
 	    fprintf(f, " ");
 	    linelen++;
 	}
-	fprintf(f, movetext);
+	fprintf(f, "%s", movetext);
 	linelen += movelen;
 
 	i++;
